sig-testing/flakiness-sla.md

```diff
@@ -0,0 +1,40 @@
+# Kubernetes Testing Flakiness SLA
+
+This document captures the expectations of the community about flakiness in
+our tests and our test infrastructure. It sets out an SLA (Service Level 
+Agreement) for flakiness in our tests, as well as actions that we will
+take when we are out of SLA.
+
+
+## Definition of a "Flake"
+
+We'll start by the definition of a _flake_. A test result is considered to
+be a flake if it both fails and passes without any changes in the code
+or environment being tested. Changes in the environment are
+notably hard to measure or control for, but we'll do our best. Changes in
+code are obviously quite easy to control.
+
+## Flakiness SLA
+We will measure flakiness based on pull requests that are run through e2e
```

Comment:
@liggitt my initial plan it to simply use historical data.

Currently it works like this:

   * User submits PR
   * PR mostly passes, but there is one flake
   * Reviewer uses @k8s-bot commands to re-run
   * Test passes

From that (which we can scrape from github data) we know that it was a flake, since the code didn't change, but the test failed, then passed.

Put another way, we're already basically double running every test. (and with the TOLERATE_FLAKES env. variable that is on, 4x running them...)